src/components/molecules/UserAuthBtn.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { NavLink, useHistory } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import firebase from '../../firebase/firebase';
+
+function UserAuthBtn() {
+  const [userLogin, setUserLogin] = useState('');
+  const history = useHistory();
+  const user = useSelector((state) => state.user);
+
+  const loginBtn = useCallback(
+    () => {
+      if (user) {
+        setUserLogin('logout');
+      } else {
+        setUserLogin('signin');
+      }
+    },
+    [user],
+  );
+
+  const handleClick = () => {
+    if (user) {
+      history.push('/');
+      return firebase.logout();
+    }
+    return loginBtn();
+  };
+
+  useEffect(() => {
+    loginBtn();
```

Comment:
Why not just put the logic of `loginBtn` here so you don't have to call a separate method?

Reply:
Excellent question. The method is used in two places (in addition to being called in the UseEffect as extra insurance that it updates properly). If the method written in the useEffect I believe that's fine for where it's called in the JSX. However the handleClick method suffers.